app/src/main/java/com/duckduckgo/app/onboarding/ui/page/WelcomePage.kt

```diff
@@ -24,28 +26,79 @@ import android.view.WindowManager
 import androidx.core.content.ContextCompat
 import androidx.core.view.ViewCompat
 import androidx.core.view.ViewPropertyAnimatorCompat
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.lifecycleScope
 import com.duckduckgo.app.browser.R
 import com.duckduckgo.app.global.view.html
 import kotlinx.android.synthetic.main.content_onboarding_welcome.*
 import kotlinx.android.synthetic.main.include_dax_dialog_cta.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
+@ExperimentalCoroutinesApi
 class WelcomePage : OnboardingPageFragment() {
 
+    @Inject
+    lateinit var viewModelFactory: WelcomePageViewModelFactory
+
     private var ctaText: String = ""
     private var welcomeAnimation: ViewPropertyAnimatorCompat? = null
     private var typingAnimation: ViewPropertyAnimatorCompat? = null
 
+    private val events = ConflatedBroadcastChannel<WelcomePageView.Event>()
+
+    private val welcomePageViewModel: WelcomePageViewModel by lazy {
+        ViewModelProvider(this, viewModelFactory).get(WelcomePageViewModel::class.java)
+    }
+
     override fun layoutResource(): Int = R.layout.content_onboarding_welcome
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
 
-        primaryCta.setOnClickListener { onContinuePressed() }
+        primaryCta.setOnClickListener { event(WelcomePageView.Event.OnPrimaryCtaClicked) }
 
         configureDaxCta()
         beginWelcomeAnimation(ctaText)
     }
 
+    override fun onStart() {
+        super.onStart()
+
+        lifecycleScope.launch {
```

Comment:
This new pattern looks great, and it remembers me of old good RxJava days :D

In any case, I would recommend that before introducing new patterns we could discuss them internally. Could you do a brief summary of this approach at our next mobile meeting? Thanks!